python-threatexchange/threatexchange/cli/config_cmd.py

```diff
@@ -158,7 +199,7 @@ def _add_argument(cls, ap: argparse._ArgumentGroup, field: Field) -> None:
             if isinstance(origin, type) and issubclass(
                 origin, collections.abc.Collection
             ):
-                argparse_type = lambda s: origin(p.strip() for p in s.split(","))  # type: ignore  # mypy not smart enough for origin == type
+                argparse_type = lambda s: origin(arg_type(p.strip()) for p in s.split(","))  # type: ignore  # mypy not smart enough for origin == type
```

Comment:
am I correct that this is the meaningful  fix/change?

Reply:
This is the only change needed to fix the core, then I added everything else for testing.